Google.Api.Generator/RoslynUtils/Modifier.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Google.Api.Generator.RoslynUtils
+{
+    [Flags]
+    internal enum Modifier
+    {
+        DontCare = 0,
+        NoModifiers = 0,
+
+        Public = 0x1,
+        Static = 0x2,
+        Private = 0x4,
+        Readonly = 0x8,
+        Async = 0x10,
+        Sealed = 0x20,
+        Partial = 0x40,
+        Abstract = 0x80,
+        Virtual = 0x100,
+        Override = 0x200,
+        Internal = 0x400,
+    }
+
+    internal static class ModifierExtensions
+    {
+        private static readonly SyntaxToken PUBLIC_TOKEN = SyntaxFactory.Token(SyntaxKind.PublicKeyword);
```

Comment:
Any reason for shouty case here?

Reply:
Not really - I can never remember the right format for these constant-like things ;)

Reply:
Done